src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullCoalescingOperator.cs

```diff
@@ -87,9 +87,10 @@ private BoundExpression MakeNullCoalescingOperator(
 
             // if left conversion is intrinsic implicit (always succeeds) and results in a reference type
             // we can apply conversion before doing the null check that allows for a more efficient IL emit.
+            // span conversion is from a reference type (array/string) to a value type ([ReadOnly]Span) so it cannot participate here.
             Debug.Assert(rewrittenLeft.Type is { });
             if (rewrittenLeft.Type.IsReferenceType &&
-                BoundNode.GetConversion(leftConversion, leftPlaceholder) is { IsImplicit: true, IsUserDefined: false })
+                BoundNode.GetConversion(leftConversion, leftPlaceholder) is { IsImplicit: true, IsUserDefined: false, IsSpan: false })
```

Comment:
>IsSpan: false [](http://example.com/codeflow?start=118&length=13)

Given the original comment above the `if`, it feels like the condition was a somewhat "roundabout" way of checking for an implicit reference conversion. With the additional condition we are still trying to check for what we are not looking for rather than checking for what we are looking for. Consider adjusting condition accordingly in order to make it more robust in long run.